sky/shell/platform/ios/framework/Source/FlutterViewController.mm

```diff
@@ -435,11 +451,37 @@ - (UIStatusBarStyle)preferredStatusBarStyle {
 
 - (void)dealloc {
   [[NSNotificationCenter defaultCenter] removeObserver:self];
+  [super dealloc];
+}
 
-  [_dynamicServiceLoader release];
-  [_dartProject release];
+#pragma mark - Application Messages
 
-  [super dealloc];
+- (void)sendString:(NSString*)message
+   withMessageName:(NSString*)messageName {
+  _appMessageSender->SendString(messageName.UTF8String, message.UTF8String,
+      [](const mojo::String& response) { });
+}
+
+- (void)sendString:(NSString*)message
+   withMessageName:(NSString*)messageName
+          callback:(void(^)(NSString*))callback {
+  base::scoped_nsprotocol<StringCallback> callback_ptr(callback);
```

Comment:
* `StringCallback` is not a protocol. `base::scoped_nsprotocol` does not make sense here.
* The possibly stack block must be moved to the heap via `Block_Copy` before a scoped pointer can assume ownership of the same.

So something like: `base::scoped_nsobject<id> callback_ptr(Block_copy(callback))`

I am not sure why this was not crashing for you when you attempted to execute the block in the callback after returning from the method.